components/FavoriteButton/index.js

```diff
@@ -0,0 +1,49 @@
+import styled from "styled-components";
+
+export default function FavoriteButton({
+  artPiecesInfo,
+  onToggleFavorite,
+  slug,
+}) {
+  const currentArtPiece = artPiecesInfo.find(
+    (artPiece) => artPiece.slug === slug
+  ) ?? { isFavorite: false };
+
+  const { isFavorite } = currentArtPiece;
```

Comment:
it would be good to have a dump component. You can think about a prop ``isFavorite`` and decide when you call the FavoriteButton to give a true or false.